packages/shared/src/hooks/useSettings.ts

```diff
@@ -67,23 +79,49 @@ export default function useSettings(
 
   useEffect(() => {
     if (remoteSettings) {
-      const lightMode = remoteSettings.userSettings.theme === 'bright';
-      setLightMode(lightMode);
-      toggleTheme(lightMode);
+      const theme = remoteSettings.userSettings.theme;
+      console.log(theme);
+      if (theme === 'auto') {
+        toggleSystemTheme(true);
+        setThemeMode('auto');
+      } else {
+        const lightMode = theme === 'bright';
+        setLightMode(lightMode);
+        setThemeMode(lightMode ? 'light' : 'dark');
+        toggleTheme(lightMode);
+        toggleSystemTheme(false);
+      }
       const cloneSettings = { ...remoteSettings.userSettings };
       delete cloneSettings.theme;
       setCachedSettings(cloneSettings);
     }
   }, [remoteSettings]);
 
   useEffect(() => {
-    const lightModeCookieValue = localStorage.getItem(themeCookieName);
-    if (lightModeCookieValue === 'true') {
-      setLightMode(true);
-      document.documentElement.classList.add('light');
+    const themeModeCookieValue = localStorage.getItem(themeModeCookieName);
+    setThemeMode(themeModeCookieValue);
+    if (themeModeCookieValue == 'auto') {
+      toggleSystemTheme(true);
+    } else {
+      const lightModeCookieValue = localStorage.getItem(themeCookieName);
+      if (lightModeCookieValue === 'true') {
+        setLightMode(true);
+        document.documentElement.classList.add('light');
+      }
     }
   }, []);
 
+  const triggerThemeChange = async (isLightMode: boolean): Promise<void> => {
+    setLightMode(isLightMode);
+    toggleTheme(isLightMode);
+    if (userId) {
+      await updateRemoteSettings({
+        ...settings,
+        theme: isLightMode ? 'bright' : 'darcula',
```

Comment:
`setSettings` has a `setCachedSettings` method though. Rather than do a logic check there, I wrapped this `updateRemoteSettings` in a function called `updateRemoteSettingsFn`. Included a triple ternary check inside this. Not sure whether it is a proper standard though. Have a look and let me know

Reply:
Update:
Fixed all the issues.
Let me know if there is anything